lib/engine/game/g_2038/step/waterfall_auction.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/waterfall_auction'
+
+module Engine
+  module Game
+    module G2038
+      module Step
+        class WaterfallAuction < Engine::Step::WaterfallAuction
+          def may_purchase?(company)
+            return false unless super
+            return true if @purchasing_first_minor
+
+            !minor?(company)
+          end
+
+          def can_auction?(company)
+            return true if @process_round_end_auction && @bids[company].size > 1
+
+            super
+          end
+
+          def min_bid(company)
+            return unless company
+            return company.min_bid if may_purchase?(company)
+
+            high_bid = highest_bid(company)
+            high_bid ? high_bid.price + min_increment : company.min_bid
+          end
+
+          def bid_str(company)
+            !auctioning && company && minor?(company) && company == @companies.first ? 'Buy' : 'Place Bid'
+          end
+
+          def placement_bid(bid)
+            @purchasing_first_minor = bid.company && bid.company == @companies.first
+            super
+            @purchasing_first_minor = false
+          end
+
+          def minor?(company)
+            @game.minors.find { |m| m.id == company.id }
```

Comment:
makes more sense to do any? since you just expect a boolean

Reply:
Ok, I was avoiding any? because I was under the impression from the wiki that Opal didn't produce good code for it but checking just now it only mentions the GC which is probably the same with this statement anyway.  Will change.